qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -126,6 +113,175 @@ def get_subset_of(self, index: Union[str, int]) -> "CurveData":
+    """
+
+    method: str = None
+    model_repr: Dict[str, str] = None
+    success: bool = True
+    nfev: int = None
+    message: str = ""
+    dof: float = None
+    chisq: float = None
+    reduced_chisq: float = None
+    aic: float = None
+    bic: float = None
+    params: Dict[str, float] = None
+    var_names: List[str] = None
+    x_data: np.ndarray = None
+    y_data: np.ndarray = None
+    covar: np.ndarray = None
+    correl: np.ndarray = dataclasses.field(init=False, repr=False)
+    ufloat_params: Dict[str, uncertainties.UFloat] = dataclasses.field(init=False, repr=False)
+
+    def __post_init__(self):
+        self._set_correlation_matrix(self.covar)
+        self._set_ufloat_params(self.covar)
+
+    @property
+    def x_range(self) -> Tuple[float, float]:
+        """Range of X values."""
+        return min(self.x_data), max(self.x_data)
+
+    @property
+    def y_range(self) -> Tuple[float, float]:
+        """Range of Y values."""
+        return min(self.y_data), max(self.y_data)
+
+    @deprecated_function("0.5", "Use '.ufloat_params' which returns a dictionary instead.")
+    def fitval(self, key: str) -> uncertainties.UFloat:
+        """Deprecated. Return UFloat parameter specified by the key."""
+        return self.ufloat_params[key]
+
+    def _set_ufloat_params(self, covar: np.ndarray):
+        """A helper function to compute ufloat parameters from covariance matrix."""
+        if self.params is None:
+            object.__setattr__(self, "ufloat_params", None)
+            return
+
+        if covar is not None:
+            ufloat_fitvals = uncertainties.correlated_values(
+                nom_values=[self.params[name] for name in self.var_names],
+                covariance_mat=covar,
+                tags=self.var_names,
+            )
+        else:
+            # Invalid covariance matrix. Std dev is set to nan, i.e. not computed.
+            ufloat_fitvals = uarray(
+                nominal_values=[self.params[name] for name in self.var_names],
+                std_devs=np.full(len(self.var_names), np.nan),
+            )
+
+        # Combine fixed params and fitting variables into a single dictionary
+        # Fixed parameter has zero std_dev
+        ufloat_params = {}
+        for name in self.params.keys():
+            try:
+                uind = self.var_names.index(name)
+                ufloat_params[name] = ufloat_fitvals[uind]
+            except ValueError:
+                ufloat_params[name] = uncertainties.ufloat(self.params[name], std_dev=0.0)
+
+        object.__setattr__(self, "ufloat_params", ufloat_params)
+
+    def _set_correlation_matrix(self, covar: np.ndarray):
+        """A helper function to compute correlation matrix from covariance matrix."""
+        if covar is not None:
+            # This is how uncertainties computes correlation matrix
+            stdevs = np.sqrt(np.diag(covar))
+            correl = covar / stdevs / stdevs[:, np.newaxis]
```

Comment:
This looks a bit odd. This is doing `Cov / (sigma * sigma.T)` right? Where `sigma` is a vector of standard deviations.

Reply:
Likely not. `np.newaxis` creates extra dimension.
```python
import numpy as np

v = np.array([1,2,3])

>>> 1 / v / v.T
array([1.        , 0.25      , 0.11111111])

>>> 1 / v / v[:, np.newaxis]
array([[1.        , 0.5       , 0.33333333],
       [0.5       , 0.25      , 0.16666667],
       [0.33333333, 0.16666667, 0.11111111]])
```